test/functional/cmdLineTests/jfr/build.xml

```diff
@@ -0,0 +1,81 @@
+<?xml version="1.0"?>
+
+<!--
+  Copyright IBM Corp. and others 2024
+
+  This program and the accompanying materials are made available under
+  the terms of the Eclipse Public License 2.0 which accompanies this
+  distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+  or the Apache License, Version 2.0 which accompanies this distribution and
+  is available at https://www.apache.org/licenses/LICENSE-2.0.
+
+  This Source Code may also be made available under the following
+  Secondary Licenses when the conditions for such availability set
+  forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+  General Public License, version 2 with the GNU Classpath
+  Exception [1] and GNU General Public License, version 2 with the
+  OpenJDK Assembly Exception [2].
+
+  [1] https://www.gnu.org/software/classpath/license.html
+  [2] https://openjdk.org/legal/assembly-exception.html
+
+  SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
+-->
+<project name="jfr" default="build" basedir=".">
+	<taskdef resource="net/sf/antcontrib/antlib.xml" />
+	<description>
+		Build cmdLineTests_jfr
+	</description>
+
+	<import file="${TEST_ROOT}/functional/cmdLineTests/buildTools.xml"/>
+
+	<!-- set properties for this build -->
+	<property name="DEST" value="${BUILD_ROOT}/functional/cmdLineTests/jfr" />
+	<property name="src" location="./src"/>
+	<property name="build" location="./bin"/>
+
+	<target name="init">
+		<mkdir dir="${DEST}" />
+		<mkdir dir="${build}" />
+	</target>
+
+	<target name="compile" depends="init" description="Using java ${JDK_VERSION} to compile the source ">
+		<echo>Ant version is ${ant.version}</echo>
+		<echo>============COMPILER SETTINGS============</echo>
+		<echo>===fork:                         yes</echo>
+		<echo>===executable:                   ${compiler.javac}</echo>
+		<echo>===debug:                        on</echo>
+		<echo>===destdir:                      ${DEST}</echo>
+		<!-- For the time being, JFR tests are only limited to JDK17+ and Linux-x86. -->
+		<if>
+			<not>
+				<matches string="${JDK_VERSION}" pattern="^(8|9|1[0-6])$$" />
```

Comment:
Besides JDK version, `Linux-x86` needs to be checked as well, other platforms won't have `VM.triggerExecutionSample()`.

Reply:
> Linux-x86 needs to be checked as well

What's the corresponding variable to check?

 > other platforms won't have `VM.triggerExecutionSample()`.

Actually, currently `VM.triggerExecutionSample()` is defined in all the platforms.

Reply:
`VM.triggerExecutionSample()` is decorated with `/*[IF JFR_SUPPORT]*/` which is enabled `ifeq (true,$(OPENJ9_ENABLE_JFR))`. `--enable-jfr` is set by default only on `xa64`.
I expect other platforms don't have this method.

>> Linux-x86 needs to be checked as well

> What's the corresponding variable to check?

There is no straightforward way like the Java version check. Maybe a system property like https://github.com/eclipse-openj9/openj9/blob/d776a4a40f493f86d9376a6e2d667f9eb68a8935/jcl/src/java.base/share/classes/java/lang/System.java#L713-L719 and https://github.com/adoptium/TKG/blob/8a5ae9e5330e8ec1bef2985600c3ee2c19605959/src/org/openj9/envInfo/JavaInfo.java#L232

Reply:
Ant has OS condition tag (https://ant.apache.org/manual/Tasks/conditions.html). I think `isUnix && !isMacos` should be good to detect Linux?

Reply:
On the other hand, I guess the ultimate way is to add a system property? Can you explain how could the test flags be used?

Reply:
Hardcoding an OS/Arch check requires ongoing change while JFR support is expanded to other platforms.
An approach based on https://github.com/eclipse-openj9/openj9/pull/19917#discussion_r1692067783 is appropriate.

Reply:
The test flags are set by TKG framework, an example is https://github.com/adoptium/TKG/blob/8a5ae9e5330e8ec1bef2985600c3ee2c19605959/src/org/openj9/envInfo/JavaInfo.java#L232

OpenJ9 test build/run according to if a flag is set
https://github.com/eclipse-openj9/openj9/blob/9142f7eb060d4a74d9def47039b06edf7a8c65c4/test/functional/cmdLineTests/criu/build.xml#L65-L70